### PR TITLE
Us20 implement state loops

### PR DESCRIPTION
**Pull request by Heso113**
I finished the loops as much as I could at this moment in time. I need a fighting system from the physics side and some pathfinding to be able to finnish up the last pieces of the pussel. I however think that this parts are small enough to consider this task pretty much done.

### CODE REVIEW

File: Game/GameMain/gameAI/AIComponents.h [hunk 1]
```diff
@@ -35,10 +36,23 @@ namespace ecs
 		};
 
 		struct MoveStateComponent : public ECSComponent<MoveStateComponent>
+		{
+			STATE goalState = STATE::IDLE;
+			DirectX::XMFLOAT3 goalPos = {5.0f, 0.0f, 0.0f};
```
Comment (Wessebanan) [inline Game/GameMain/gameAI/AIComponents.h]: varför default 5,0,0?

File: Game/GameMain/gameAI/AISystems.h [hunk 2]
```diff
@@ -87,21 +133,173 @@ namespace ecs
...
+					ecs::ECSUser::removeComponent(entity.entity->getID(), ecs::components::MoveStateComponent::typeID);
+					switch (move->goalState)
+					{
+					case LOOT:
+						ecs::ECSUser::createComponent(entity.entity->getID(), loot_state);
+						break;
+					default:
+						ecs::ECSUser::createComponent(entity.entity->getID(), atk_state);
+						break;
+					}
```
Comment (Wessebanan) [inline Game/GameMain/gameAI/AISystems.h]: Borde inte default vara idle?

Comment (Heso113) [reply Game/GameMain/gameAI/AISystems.h]: Fixed a case for Attack and default to idle. It should never go to idle from this state but it is probably better to have idle as default as a safety measure.

File: Game/GameMain/gameAI/AIComponents.h (hunk 1 above)
Comment (Heso113) [reply Game/GameMain/gameAI/AIComponents.h]: Changed it to default to 0, 0, 0 just to avoid getting garbage values if someone forgets to set it.

**Pull request merged**